jax/_src/basearray.pyi

```diff
@@ -186,3 +186,24 @@ class Array(abc.ABC):
   def at(self) -> Any: ...
   @property
   def weak_type(self) -> bool: ...
+
+  # Methods defined on ArrayImpl, but not on Tracers
+  def addressable_data(self, index: int) -> Array: ...
+  def block_until_ready(self) -> Array: ...
+  def copy_to_host_async(self) -> None: ...
+  def delete(self) -> None: ...
+  def device(self) -> Device: ...
+  @property
+  def device_buffer(self) -> Array: ...
+  @property
+  def device_buffers(self) -> Sequence[Array]: ...
```

Comment:
Do we want to expose these? These methods are deprecated `device_buffer` and `device_buffers`

Reply:
Makes sense, yeah we should probably leave these out. Are there any other deprecated methods/properties in this list?

Reply:
Rest of them are good!